### ISSUE TITLE
Title Search seems to return wrong results

### ISSUE THREAD

**Issue by bsavant** [laurent22/joplin#10549]
### Operating system

Linux

### Joplin version

3.0.6

### Desktop version info

Joplin 3.0.6 (prod, linux)

Client ID: b2bc7fbd98864f42af824c2361a1a1bb
Sync Version: 3
Profile Version: 47
Keychain Supported: No

Revision: 18b9f5c

Backup: 1.4.1

### Current behaviour

There is a marked difference between the results that a text search returns vs a title search.

1. Create a note with "2024-05-06" in the title.
2. Create a note with "2024-06-05" in the title.
3. Search for:  "2024-06-05", note results
4. Search for: title: "2024-06-05", note results

### Expected behaviour

As expected, the search for:  "2024-06-05" should return the 2024-06-05 note only.

When searching for: title: "2024-06-05", I would expect only to get back 2024-06-05 as well, but it also is returning the 2024-05-06 note.

https://github.com/laurent22/joplin/assets/209364/4f4c023c-cffa-46bc-a380-366d6a67b3f9

**Comment by laurent22** [comment]
Yes looks like a bug. I'd be fine with it if the searched date was the first result but it's not, so it's probably so incorrect filtering

**Comment by suzitsam** [comment]
@laurent22 I would like to work on this.

**Comment by arjun-kshirsagar** [comment]
Hi @laurent22 I have started working on the code, plz assign this issue to me or accept the one who has made the first PR.
